core/arch/arm/kernel/stmm_sp.c

```diff
@@ -659,36 +796,69 @@ static bool stmm_handle_storage_service(struct thread_svc_regs *regs)
 			 TEE_DATA_FLAG_ACCESS_WRITE |
 			 TEE_DATA_FLAG_SHARE_READ |
 			 TEE_DATA_FLAG_SHARE_WRITE;
-	uint32_t action = regs->x3;
-	void *va = (void *)regs->x4;
-	unsigned long len = regs->x5;
-	unsigned long offset = regs->x6;
+	uint32_t action = 0;
+	void *va = NULL;
+	unsigned long len = 0;
+	unsigned long offset = 0;
 	char obj_id[] = FILENAME;
 	size_t obj_id_len = strlen(obj_id);
 	TEE_Result res = TEE_SUCCESS;
+	uint32_t stmm_rc = STMM_RET_INVALID_PARAM;
+
+#ifdef ARM64
+	action = regs->x3;
+	va = (void *)regs->x4;
+	len = regs->x5;
+	offset = regs->x6;
 
 	switch (action) {
 	case FFA_SVC_RPMB_READ:
 		res = sec_storage_obj_read(TEE_STORAGE_PRIVATE_RPMB, obj_id,
 					   obj_id_len, va, len, offset, flags);
-		service_compose_direct_resp(regs, res);
-
-		return true;
+		stmm_rc = tee2stmm_ret_val(res);
```

Comment:
This is a bugfix of present code. Could be nice to mention it in the commit message.

Reply:
ok.